File: examples/typescript-example/src/ui/userList.ts
// UI module that depends on services (cross-layer dependency for testing)
import { UserService } from '../services/user';
import { Logger } from '../utils/logger';

export class UserListPage {
  private userService: UserService;
  private logger: Logger;

  constructor() {
    this.userService = new UserService();
    this.logger = new Logger('UserListPage');
  }

  renderUsers(): string {
    const users = this.userService.getAll();
    return `<ul>${users.map((u) => `<li>${u.name}</li>`).join('')}</ul>`;
  }
}

export function formatUser(name: string): string {
  return `User: ${name}`;
}

File: examples/python-example/src/ui/user_list.py
"""User list page - depends on services (cross-layer)."""
from src.services.user import UserService
from src.utils.logger import Logger


def format_user(name: str) -> str:
    return f"User: {name}"


class UserListPage:
    def __init__(self):
        self.user_service = UserService()
        self.logger = Logger("UserListPage")

    def render_users(self) -> str:
        users = self.user_service.get_all()
        return "<ul>" + "".join(f"<li>{u['name']}</li>" for u in users) + "</ul>"
